manifests/init.pp

```diff
@@ -54,25 +55,27 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  $rh_user                           = undef,
-  $rh_password                       = undef,
-  $org                               = undef,
-  $activationkey                     = undef,
-  $pool                              = undef,
-  $proxy_hostname                    = undef,
-  $proxy_port                        = undef,
-  $proxy_user                        = undef,
-  $proxy_password                    = undef,
-  $baseurl                           = 'https://cdn.redhat.com',
-  $servername                        = 'subscription.rhsm.redhat.com',
-  $serverprefix                      = '/subscription',
-  $serverport                        = 443,
-  $ca_cert_dir                       = '/etc/rhsm/ca/',
-  $repo_ca_cert_filename             = 'redhat-uep.pem',
-  $repo_ca_cert_source               = undef,
-  $manage_repos                      = 1,
-  $full_refresh_on_yum               = 0,
-  $package_ensure                    = 'latest',
+  Optional[String] $rh_user = undef,
```

Comment:
back in the days it was not converted to Puppet 4+ style because Sat6 did deliver Puppet 3.6 only. 

So adding Puppet types would break anybody still using it with 3.6.

Reply:
Since [Satellite 6.3 Puppet 4 is supported](https://access.redhat.com/documentation/en-us/red_hat_satellite/6.3/html/release_notes/chap-red_hat_satellite-release_notes-new_features_and_enhancements)

Reply:
And Satellite 6.2 is going EOL in May 2019. I highly recommend people to upgrade.